src/System.Private.CoreLib/shared/System/Number.IntegerFormatting.cs

```diff
@@ -0,0 +1,1541 @@
+                UInt32ToIntegerBuffer(value, ref integer);
+                ValueStringBuilder sb;
+                unsafe
+                {
+                    char* stackPtr = stackalloc char[CharStackBufferSize];
+                    sb = new ValueStringBuilder(new Span<char>(stackPtr, CharStackBufferSize));
+                }
+                if (fmt != 0)
+                {
+                    IntegerBufferToString(ref sb, ref integer, fmt, digits, info);
+                }
+                else
+                {
+                    IntegerBufferToStringFormat(ref sb, ref integer, format, info);
+                }
+                return sb.TryCopyTo(destination, out charsWritten);
+            }
+        }
+
+        public static bool TryFormatUInt64(ulong value, ReadOnlySpan<char> format, IFormatProvider provider, Span<char> destination, out int charsWritten)
+        {
+            // Fast path for default format
+            if (format.Length == 0)
+            {
+                return TryUInt64ToDecStr(value, digits: -1, destination, out charsWritten);
+            }
+
+            char fmt = ParseFormatSpecifier(format, out int digits);
+            char fmtUpper = (char)(fmt & 0xFFDF); // ensure fmt is upper-cased for purposes of comparison
+            if ((fmtUpper == 'G' && digits < 1) || fmtUpper == 'D')
+            {
+                return TryUInt64ToDecStr(value, digits, destination, out charsWritten);
+            }
+            else if (fmtUpper == 'X')
+            {
+                // The fmt-(X-A+10) hack has the effect of dictating whether we produce uppercase or lowercase
+                // hex numbers for a-f. 'X' as the fmt code produces uppercase. 'x' as the format code
+                // produces lowercase.
+                return TryInt64ToHexStr((long)value, (char)(fmt - ('X' - 'A' + 10)), digits, destination, out charsWritten);
+            }
+            else
+            {
+                NumberFormatInfo info = NumberFormatInfo.GetInstance(provider);
+                IntegerBuffer integer = default;
+                UInt64ToIntegerBuffer(value, ref integer);
+                ValueStringBuilder sb;
+                unsafe
+                {
+                    char* stackPtr = stackalloc char[CharStackBufferSize];
+                    sb = new ValueStringBuilder(new Span<char>(stackPtr, CharStackBufferSize));
+                }
+                if (fmt != 0)
+                {
+                    IntegerBufferToString(ref sb, ref integer, fmt, digits, info);
+                }
+                else
+                {
+                    IntegerBufferToStringFormat(ref sb, ref integer, format, info);
+                }
+                return sb.TryCopyTo(destination, out charsWritten);
+            }
+        }
+
+        internal static unsafe void IntegerBufferToString(ref ValueStringBuilder sb, ref IntegerBuffer integer, char format, int nMaxDigits, NumberFormatInfo info)
```

Comment:
The `ref NumberBuffer number` parameters for the integer case where changed to `ref IntegerBuffer integer`